pkg/sql/catalog/tabledesc/validate.go

```diff
@@ -454,6 +454,83 @@ func (desc *wrapper) validateInboundTableRef(
 		backReferencedTable.GetName(), by.ID)
 }
 
+// validateFK asserts that references to desc from inbound and outbound FKs are
+// valid.
+func (desc *wrapper) validateSelfFKs() error {
+	colsByID := catalog.ColumnsByIDs(desc)
+
+	// Validate all FKs where desc is the origin table.
+	// We'll validate the referenced table elsewhere.
+	for _, wrapper := range desc.OutboundForeignKeys() {
```

Comment:
It might be useful to include the index in the error messages to help with debugging.